dev/ComboBox/ComboBoxHelper.cpp

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <common.h>
+#include "ComboBoxHelper.h"
+#include "ComboBoxHelper.properties.h"
+#include "DispatcherHelper.h"
+#include "Converters.h"
+
+using CornerRadiusFilterType = CornerRadiusFilterConverter::FilterType;
+
+static constexpr auto c_popupBorderName = L"PopupBorder"sv;
+static constexpr auto c_editableTextName = L"EditableText"sv;
+GlobalDependencyProperty ComboBoxHelper::s_DropDownEventRevokersProperty{ nullptr };
+
+ComboBoxHelper::ComboBoxHelper()
+{
+    EnsureProperties();
+}
+
+void ComboBoxHelper::EnsureProperties()
+{
+    if (!s_DropDownEventRevokersProperty)
+    {
+        s_DropDownEventRevokersProperty =
+            InitializeDependencyProperty(
+                L"DropDownEventRevokers",
+                winrt::name_of<winrt::IInspectable>(),
+                winrt::name_of<winrt::ComboBoxHelper>(),
+                true /* isAttached */,
+                ValueHelper<winrt::IInspectable>::BoxedDefaultValue(),
+                nullptr);
+    }
+
+    ComboBoxHelperProperties::EnsureProperties();
+}
+
+void ComboBoxHelper::ClearProperties()
+{
+    s_DropDownEventRevokersProperty = nullptr;
+}
+
+
+// Normal ComboBox and editable ComboBox have different CornerRadius behaviors.
+// Xaml is not lifted yet when we implementing this feature so we don't have access to ComboBox code.
+// Creating this attached property to help us plug in some extra logic without touching the actual ComboBox code.
+void ComboBoxHelper::OnApplyDynamicCornerRadiusPropertyChanged(
+    const winrt::DependencyObject& sender,
+    const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (auto comboBox = sender.try_as<winrt::ComboBox>())
+    {
+        if (auto revokersInspectable = comboBox.GetValue(DropDownEventRevokersProperty()))
```

Comment:
Optional: you could remove the requirement to call RevokeAll and instead just add:
    comboBox.SetValue(DropDownEventRevokersProperty(), nullptr);
as an else clause to the below "if (shouldMonitorDropDownState)"

Though it is possible that doing it the way you have it is more explicit and hence easier to interpret.

Reply:
The current way is probably easier to understand. So leave it.

---
In reply to: [303558812](https://github.com/microsoft/microsoft-ui-xaml/pull/1023#discussion_r303558812) [](ancestors = 303558812)